api/src/paths/dwc/submission/{datasetId}/related.ts

```diff
@@ -80,12 +90,24 @@ export function getRelatedDatasetsByDatasetId(): RequestHandler {
       await connection.open();
 
       const submissionService = new SubmissionService(connection);
+      const securityService = new SecurityService(connection);
 
       const datasets = await submissionService.findRelatedDatasetsByDatasetId(datasetId);
 
+      const datasetsWithSupplementaryData = await Promise.all(
+        datasets.map(async (dataset) => {
+          const isDatasetPendingReview = await securityService.isDatasetPendingReview(dataset.datasetId);
+
+          return {
+            ...dataset,
+            supplementaryData: { isPendingReview: isDatasetPendingReview }
+          };
+        })
+      );
+
```

Comment:
Discussed with Anissa, but never wrote down. So adding here just in case:

Its not clear, but I think we can assume based on the frontend logic, that we don't want the API to reveal any information about the artifact if it is in a pending state, to non-admin users.

So this supplementary data section, should really only return `isPendingReview` if the calling user is an admin, and return an empty object if they are a non admin. The openapi and frontend types/interfaces would need to be updated accordingly of course, to make `isPendignReview` optional.

This same logic should also apply to our other Artifact endpoint that returns the `SECURE`, `UNSECURE`, `PENDING` statuses. If the user is an admin, no change. But if they are a non admin, we should really only return `SECURE` or `UNSECURE` (if the status is PENDING, we would return SECURE instead). This exact logic is handled this way on the frontend, but really should be a backend task, and the frontend should just display the status without having to check what role the user has. SO the backend will get slightly more complicated, but the frontend will get simpler, and overall the API response data is more secure.

The only other recommendation is for both of the above changes, we should move the code that builds the supplemental section into a new service function. It can still be called in a map at this level, and added to the response, but if we start to add new supplemental fields to the response over time, we ideally don't want to keep having to modify the endpoint.